packages/default/scss/checkbox/_layout.scss

```diff
@@ -159,10 +169,22 @@
             margin-left: 0;
             margin-right: $checkbox-label-margin-x;
         }
+        .k-checkbox + kendo-label.k-checkbox-label,
+        .k-checkbox-label + kendo-label.k-checkbox {
+            margin-left: 0;
+            margin-right: calc( #{$checkbox-label-margin-x} - 4px );
+        }
         .k-checkbox-label > .k-checkbox {
             margin-right: 0;
             margin-left: $checkbox-label-margin-x;
         }
+        kendo-label.k-checkbox-label > .k-checkbox {
+            margin-left: 0;
+        }
+        kendo-label.k-checkbox-label > .k-label {
+            margin-right: 0;
+            margin-left: $checkbox-label-margin-x;
+        }
```

Comment:
That related to the second comment. Remove or rework to handle the cases of checkbox being first or last.
```suggestion
```